### ISSUE TITLE
Bridge not responding

### ISSUE THREAD

**Issue by Vuille** [Hyrules/WinHue3#51]
WinHue 3 beta 24 regularly crashes. I get messages "Bridge not responding". 
Although the IP adresse showing for the bridge is correct, it is impossible to pair it.
I have tried several times to reinstall WinHue, without success. Bridge responds OK I use RESTful interface.
The only solution is to reset the bridge to its factory settings.
Please help!

**Comment by Hyrules** [comment]
Reinstalling won't fix anything in this case if this is a bug with the application. I will need a log file to help debug the application.

**Comment by Vuille** [comment]
I intended to attach my log file to my message, but xml type files are not accepted. How can I send the log file?

**Comment by Hyrules** [comment]
Simply rename the xml file extension to txt and it will work. Otherwise you can zip the file.

**Comment by Vuille** [comment]
Thank you. Please find the file attached.
[log 2016-12-28.txt](https://github.com/Hyrules/WinHue3/files/677646/log.2016-12-28.txt)

**Comment by Hyrules** [comment]
Since you know how to use the bridge restful interface I will need more info as the log file only tells me that there was an error deserializing your bridge store. I will need a dump of your bridge objects. I just need the list of your bridge datastore. I do not need the settings (config) of the bridge. To get it you need to a send a get request at http://<your.bridge.ip.here>/api/<your api key here> you can remove the config section of the dump if you want and send me the txt file here.

**Comment by Vuille** [comment]
The requested file is attached. It is the full command response of the bridge to the GET request to http://<bridge_IP>/api/<username>
[Bridge_objects_29dec16.txt](https://github.com/Hyrules/WinHue3/files/677800/Bridge_objects_29dec16.txt)

**Comment by Hyrules** [comment]
Thanks I will take a look at this tonight. I will find what is causing the deserialization error.

**Comment by Hyrules** [comment]
Can you try this version and let me know if you still experience this error. I have tried your dump in my test unit and it does not cause an exception. I might have modified something already.

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/677924/WinHue.3.BETA.25.Setup.zip)

**Comment by Vuille** [comment]
Thanks a lot. I have installed WinHue beta 25 and made a few tests. At first glance, everything works fine. Since it's late here (11:25 pm) I will go to bed now and perform further tests tomorrow.
Thanks again and good night.
Jean-Pierre Vuille

**Comment by Hyrules** [comment]
Nice to hear. I'll let the issue open so you can add anything else if needed. Note that the ResourceLink creator in the beta 25 does not work so it's no use to go in there and try it.

**Comment by Hyrules** [comment]
Here try this version instead. There was a bug I forgot to correct in the Schedule creator that prevented from changing settings.

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/677981/WinHue.3.BETA.25.Setup.zip)

**Comment by Vuille** [comment]
Sorry, bad news. I've installed the last version you've sent me. It worked fine for a couple of keystrokes (turning on one light or the other), then it crashed "bridge not responding". I powered the bridge off and on, then restarted the app. Everything OK for a couple of operations, then it crashed again.
Other information: from the very beginning I get the attached warning upon starting the app.
Thanks for your help.
[winhue_warning](https://cloud.githubusercontent.com/assets/24757801/21565642/e570ab7a-ce9a-11e6-8f3a-68f4e491c213.jpg)

**Comment by Hyrules** [comment]
Yes this is quite normal. If you bridge is not up to date you will encounter some problem with it. My API is always up to date so older version of the bridge might crash. You have to update your bridge to the latest version to ensure that you do no encounter deprecated problems. You should have a update button in WinHue beside the bridge setting. You can press on it an it will update the bridge to the latest API and firmware version. You won't lose anything in the process.

**Comment by Vuille** [comment]
I do not see any other button on the bridge than "restore factory settings". Is it what you mean?

**Comment by Hyrules** [comment]
In WinHue there is a button to update the bridge to the latest firmware. Right beside the bridge settings button on the main bar. Yellow exclamation mark

**Comment by Vuille** [comment]
Sorry... can't find it. I do not see any exclamation mark. Right beside the bridge settings button comes "Events log" on the right and "Resource link creator" of the left.

**Comment by Hyrules** [comment]
Is your bridge registered on the philips hue portal ?

**Comment by Vuille** [comment]
I do not think so, but  not sure

**Comment by Hyrules** [comment]
from the last dump you sent me I can say that your firmware version API is 4 version behind. You are at 1.12.0 and the latest is 1.16.0. WinHue will most likely never work properly unless your get the latest update. Also you do not seem to have any incoming connections from philips. Portail service should be true and in your case it's not. You will have to create an account at philips in order to get the updates : https://www.meethue.com/en-us/register

**Comment by Vuille** [comment]
OK, thanks, I will proceed.

**Comment by Vuille** [comment]
Hello again,
I've created an account with Philips. Now portal service is true and Portalstate signed on, incoming and outgoing are all true. On WinHue settings, I can see that a new version is available for the firmware but I do not know how to download it. Can you help me, please?

**Comment by Hyrules** [comment]
It will take a couple of hours maybe less for the bridge to get the update. There is nothing you can do meanwhile. Just wait until the update is ready to be applied. WinHue will show a Yellow Exclamation mark button beside the Bridge Settings Button when the update is ready to be applied. restart to program often so you can see if the update is ready or poll the bridge using the restful api.

**Comment by Vuille** [comment]
Please ignore my last comment. I found by myself how to update the bridge firmware, which is now 01036659. API version is 1.16.0. I trust everything is OK.
Are you Pascal Pharand?
Than you for you help and have a Happy New Year

**Comment by Hyrules** [comment]
Yes. I'm the only programmer. You can now try the program and let me know if there is are still errors while using it.

**Comment by wo-rasp** [comment]
Issue solved, Beta 25 starts nicely centered, the maximize button expands the window to max screen size.
Thanks.
There is another wish for handling of windows, plus several requests for enhancements.
I will explain it based on the property window.
Floating the window for the 1st time after start of the application will position the window at the upper left corner. Main reason for floating the property window is to increase the size of the window frame in order to be able to read all property dialogs. It would be better to float the window with approximately 25% of the application window size at the lower left corner. The reason for coosing the lower left corner is the minimum interference with other parts of the GUI.
The data shown in the property window is not updated when changed in the GUI (for example changing the brightness for a lamp using the slider will not be updated in the property window, you do need to change for example from lamp to a group of lamps and baack to the lamp to get the updated property value for brightness.
Once the floated property window is positioned, floating and docking is working as expected (e.q. the floated window is always positioned at the same position and with the same size.
However maximizing the property window and restoring it to its previous size will only work as expected until the property window is docked; floating it again will maximize the property window. When you reposition this maximized window you can monitor another underlying issue, it seems that the borderwidth of the window plus the pixels for the window title are not catered for (e.q. the window size is larger than expected, you will easily spot what I mean when you reposition the window on your screen and maximize and restore the window, you will notice that the maximized window will center on the screen, but the restored window will lineup with the upper right corner on the screen and the left and lower part of the restored window will not be visible on the screen.

**Comment by Hyrules** [comment]
Enhancement noted. I will see what I can do about them. Some of them I already know like the properties not updating. For the property windows I see the problem with the docking. This is a bug in WPF Toolkit. I have updated the library and this seems to be fixed in the new version.

**Comment by Vuille** [comment]
I'm afraid the problem is not sold. I have installed the latest bridge firmware update (01036659). Then there was also an update for the lights. API version is now 1.16.0. Meanwhile, I never saw any yellow exclamation mark on the main bar. Nevertheless, WinHue continues crashing after a couple of keystrokes, I can't use it as it is yet, sorry.

**Comment by Hyrules** [comment]
Can you send me another dump of the bridge after the update ? I will take a look at it and see if there is still a bug in the program somewhere. I have seen the bridge not responding here too but its really a rare occurence. Why is it occuring on your pc often I don't know.

**Comment by Vuille** [comment]
There it is, bridge only. If you need all objects, just tell me. 
Thanks again, I appreciate your availability.
[bridge_31dec16.txt](https://github.com/Hyrules/WinHue3/files/679681/bridge_31dec16.txt)

**Comment by Hyrules** [comment]
That last dump does not contains the bridge objects just the config...

**Comment by Vuille** [comment]
Sorry, there it is now.
[Bridge_objects_31dec16.txt](https://github.com/Hyrules/WinHue3/files/679683/Bridge_objects_31dec16.txt)

**Comment by Hyrules** [comment]
Nope it must be something else. I will take a look at the threading in the program because I suspect this is the problem.

**Comment by Hyrules** [comment]
Can you try this version and let me know if it fixes the bug ?

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/679815/WinHue.3.BETA.25.Setup.zip)

**Comment by Vuille** [comment]
No, I am really sorry. I've installed that version. It worked OK for about 7 minutes during which I checked the response of all liked objects, then it crashed "Bridge not responding".
Please find below the bridge objects file and the log file. I wish you can fix it.
Kind regards,
Jean-Pierre Vuille

[log 2017-01-01.txt](https://github.com/Hyrules/WinHue3/files/680057/log.2017-01-01.txt)
[Bridge_objects_01jan17.txt](https://github.com/Hyrules/WinHue3/files/680058/Bridge_objects_01jan17.txt)

**Comment by Hyrules** [comment]
When the bridge stop responding is it only the application that bugs or is it the whole bridge that stops to responding ? If you just restart WinHue does it works again or you have to restart both ?

**Comment by Vuille** [comment]
The bridge never stops working, I can always access it through the API RESTful interface. Only WinHue  application stops. If I close down and restart WinHue, the bug persists. Now, if I power the bridge off and on again, then WinHue restarts working and the bridge responds for a few minutes and a couple of mouse clicks, then crashes again.

**Comment by Hyrules** [comment]
I have a hunch on this... Can you try this : Make WinHue work. Close the application and send me your settings file. It's located in the C:\Users\<your user>\AppData\Roaming\WinHue. Note that the appdata folder is hidden. To get there easily type %APPDATA% in the windows explorer bar. Next play with WinHue until it crashes and send me the same settings file. So you will send me 2 files one before and one after. I want to see if anything in the settings changes.

**Comment by Vuille** [comment]
OK, I'll do this

**Comment by Vuille** [comment]
There are 3 files:
WinHueSettings-1.txt: just upen opening the application (after restarting the bridge)
WinHueSettings-2.txt: upon crashing, before closing 
WinHueSettings-3.txt after closing

[WinHueSettings-1.txt](https://github.com/Hyrules/WinHue3/files/680084/WinHueSettings-1.txt)
[WinHueSettings-2.txt](https://github.com/Hyrules/WinHue3/files/680086/WinHueSettings-2.txt)
[WinHueSettings-3.txt](https://github.com/Hyrules/WinHue3/files/680088/WinHueSettings-3.txt)

**Comment by Hyrules** [comment]
It's not that. The 3 files are the same. There must be something else crashing the application. Bridge not responding error normally due to a timeout when contacting the bridge. You are the first to report me this problem. Do you have any kind of firewall on your computer or antivirus that could block WinHue after a certain amount of time for any reason ?

**Comment by Vuille** [comment]
My PC runs under Windows 10. Windows firewall is off. My only firewall is Avast. I will turn it off for 10 minutes and see what happens.

**Comment by Vuille** [comment]
I have stopped the Avast firewall for 10 min and restated the app: crash after 2 min !... So, it's not that.

**Comment by Hyrules** [comment]
Beside the fact that the Bridge is not responding the log is perfectly normal. Bridge not responding is a normal error when the bridge is not responding and not a crash per se. Now why it's not responding is rather odd. What is causing it to not respond in the application is even odder. I'll continue to look in the code to see if I miss something but it's like trying to find a needle in a haystack. Try this version. It will not fix the problem but it will give me more info on the error. Just send me the event log. I don't need the dump. Also just a note : don't use the Bridge Pairing Form. I'm currently rewriting it and it will not work properly.


[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/680140/WinHue.3.BETA.25.Setup.zip)

**Comment by Vuille** [comment]
Please note that I have numbered the different versions of WinHue 25 Beta you have supplied me with. I have named that last one 25.4 Beta.
I have installed it. Once the app is open, it does not find any bridge; besides, button Manual IP is greyed. I selected Scan, bridge was found but the pairing process does not start. I pressed Done to exit and got an error message "Sorry but an unexpected exception occurred ..."

**Comment by Hyrules** [comment]
Yes the pairing windows is not functional. You cannot use it to pair a bridge. I will have to finish it before we can continue then.

**Comment by Vuille** [comment]
Understood. Then I cannot continue any longer.
Meanwhile, I have installed WinHue version 25-3 on another computer in my LAN: same behavior, bridge not responding after a few minutes.
By the way, where are you located? just to know how much time difference we have. I am in Switzerland, near Lausanne.

**Comment by Hyrules** [comment]
I'm near Ottawa in Canada.  I'll try to finish the windows quickly and come back to you.

**Comment by Hyrules** [comment]
Here is the latest version with properly working pairing form. I have done nothing to fix your issue here. I just want to see the log if any error occurs.

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/682988/WinHue.3.BETA.25.Setup.zip)

**Comment by Vuille** [comment]
Thank you. I will install it right away and let you know by tomorrow.

**Comment by Vuille** [comment]
App installed OK, but pairing process failed. After pressing the link button, I get message "pairing complete" but the "Done" button remains grey. impossible to close the window.

**Comment by Hyrules** [comment]
You have to set the default bridge with the default button. You won't be able to press the done button unless the pairing and the default are set. You can always close the window by pressing the X in the upper right corner.

**Comment by Vuille** [comment]
Sorry, I've tried all possible manners "Pair" or "Default" first then the other button, in all cases the "Done" button remains grey.
Besides, my bridge is not found automatically, I have to give manually its IP.

**Comment by Hyrules** [comment]
This version should work correctly.

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/683460/WinHue.3.BETA.25.Setup.zip)

**Comment by Vuille** [comment]
OK, thank you, it works. I will play with it tomorrow and let you know. If the bridge stops responding, do you want me to send a log then restart the app aso? How many logs?

**Comment by Hyrules** [comment]
No I just need the log after it crashed. No need to restart anything.

**Comment by Vuille** [comment]
There it is. App crashed after about 10 mouse clicks, just turning the lights on and off.
[log 2017-01-04.txt](https://github.com/Hyrules/WinHue3/files/684220/log.2017-01-04.txt)

**Comment by Hyrules** [comment]
I have to add some things to the log. It's not displaying variables properly. I will send you another version later today to test.

**Comment by Vuille** [comment]
OK, I wait for it with pleasure.

**Comment by Hyrules** [comment]
Here we go. Let's try this one.

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/686063/WinHue.3.BETA.25.Setup.zip)

**Comment by Hyrules** [comment]
The error is as I suspected. It's a timeout error. Meaning that the application sends the message and the bridge does not reply in a timely manner. I'm still not sure if the problem is the app or local to your network. Even if you can reproduce the problem on another machine doesn't mean it is the app. I can be something with the local network. You tell me that the application does not detect the bridge when you do a detection, this means that there is something filtering upnp request or the bridge takes more than 5 seconds to reply which is not normal. My first idea would be to increase the timeout value and see if this helps. I will add the possibility to change the timeout value tonight in the application settings. Right now you can only change the upnp timeout value and it's not the value you need to change you could try changing and see if the bridge is detected. It could give a clue on the time out. If not then I will try changing the system that sends the web request for an already supported and existing api.

**Comment by Vuille** [comment]
Thank you for your efforts. Is there anything to do about the port? Which port is being used to communicate with the bridge on the network? Shall I redirect it?

**Comment by Hyrules** [comment]
By default the port is 80. it's the standard web port used for internet communication. This ports should not normally be changed or redirected. Otherwise there could be some sort of filtering by firewall on computers or routers. A proxy could also intefere.

**Comment by Vuille** [comment]
I have no idea how to change the UPnP timeout value. I looked around in the settings and the documentation of the router but did not find anything adequate. I need to wait tomorrow and request support from the ISP people. Meanwhile, I just rebooted the router.
If the problem is in the network, how do you explain that the bridge always responds to the RESTful API interface?

**Comment by Hyrules** [comment]
The Upnp timeout value is located in WinHue settings form. You do not need to change anything in your router. As for the bridge restful interface I really don't know. As I said you are the only one right now with this problem at the moment and the first to report it so it means that the application is working ok for most of the people. There must be something specific in your setup that causes that problem. I will try something else tonight. I have found a way to change the minimum number of concurrent connections in c# by default it's set to 2 I will make this a bigger number and see if the problem persists. If this does not to it I will try to change WinHue restful code for a library that is supported and already know to work ok. If this  still does not fix the problem I will have to rework the whole main form view which is what is going to happen one way or another in the near future because i'm converting the app to a mvvm programmation paradigm.

**Comment by Hyrules** [comment]
Here is another version to try. I have increased the number of connections from 2 to 20 and i have added a timeout settings in the application settings windows so you can try a bigger number. Value is in milliseconds and 1000 milliseconds = 1 seconds so by defaut it's 3000ms or 3 seconds.

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/688614/WinHue.3.BETA.25.Setup.zip)

**Comment by wo-rasp** [comment]
Vuille, 
There must be something in your network that differs from standard setups.
It certainly helps Hyrules to fix nifty and subtle issues.
It is not normal behaviour that the bridge is not automatically found (also will help others to find underlying issues and problems)
If you want to get to the root of the timeout problem you do need to track the traffic on your network for example using wireshark (although it helps to fix underlying issues and problems adding of workarounds may not be beneficial for most users and trigger other side effects).

**Comment by Hyrules** [comment]
i've seen as much as 60% reduction of code in certain view models and views. This means less variable and mostlikely a smaller memory footprint. Also less chances of having bugs. The current paradigma is also some rather crude implementation of a mvvm with views and modelviews joint together. I'm reworking it to rework proper mvvm paradigma and also proper WPF validation instead of in code validation and GUI defined Datacontext instead of code defined. In all my experience with forms that are already converted they seems to react quicker and act snappier. In ther words I am reimplementing properly what I had already implemented.

**Comment by Vuille** [comment]
The fact is the latest version does not solve the problem. I get the same error "Bridge not responding" after a few mouse clicks.
I will follow your recommendation and see what wireshark says. I'll let you know.

**Comment by wo-rasp** [comment]
@Hyrules 
You are right, I forgot to assess that with C# you may be using use the complete Microsoft toolchain and not a J2EE environment. Sounds reasonable to me.
@Vuille 
Will be interessting to see where communication locks up.
@Hyrules 
Could it be that the mismatch for the OSRAM Lamp is causing the problem ?
In the Config file:
- the model ID is stated as "Classic B40 TW", but support is only defined for the "Classic A60 TW".
- the type is configured as "Color temperature light", but support is defined as "Color Temperature Light"

**Comment by Hyrules** [comment]
Interesting I would be suprise but at this point any test will help. Anyway you can correct this easily by modifiying the lamp file in the Winhue folder. Editing it with at text editor will allow to correct this.

**Comment by Vuille** [comment]
This time, I believe we get to the target! I have removed the OSRAM lamp and I have no more errors. I did not play for a long time, anyway much longer than these last days. I am also quite sure the problem occurred after I inserted the OSRAM lamp in the system. Please note that the problem only occurs with WinHue. That lamp works OK and does not create any problem when I manage the system with the RESTful API interface
I have read somewhere how to make an OSRAM lamp really compatible with Philips Hue. The process requires a Lightify bridge. Since I own one, I will try tomorrow and let you know.

**Comment by Hyrules** [comment]
Nice catch @wo-rasp I will see if I can fix this because you might not be the only one using this type of light. It's odd that WinHue detects this light as a Classic A60 TW when it's a Classic B40 TW - LIGHTIFY. I'll correct the detection pattern and add the light to the list so it works correctly.

**Comment by Vuille** [comment]
Just another comment: in the last version of the app. the "Bridge Settings" menu does not show any data.

**Comment by Hyrules** [comment]
Normal i'm currently rewriting the form.

**Comment by wo-rasp** [comment]
I do not own one, i detected it in Vuille's log, and there you can associate the fact that the brdge stops responding when handling Device #3.
I assume that on the REST API level an "A60 TW" is compatible with a "B40 TW" as both share the same features except: the E27/E14 socket, the physical shape, the lumen/power consumption.

**Comment by Hyrules** [comment]
I was talking to Vuille but after looking at my phrase again yes It might have been looking like I was addressing you by mistake. But thanks for helping finding this.

**Comment by Vuille** [comment]
Hello, While you are rewriting WinHue application to accommodate OSRAM lamps with E14 sockets, I wish you could also take care of the  OSRAM Plug. Lamps with E14 sockets and that plug are definitely missing in the offer of Philips Hue. Based on Philips declarations, they are Hue compatible.

**Comment by Hyrules** [comment]
Sure. I will see what I can do about them.

**Comment by Vuille** [comment]
Hello,
When do you plan to issue a version of WinHue compatible with OSRAM plug and E14 sockets lamps?

**Comment by Hyrules** [comment]
Unfortunately it's not quite ready yet. I really don't know when I'll release a new version. I'm still not finished with the main form rewrite and I have yet to try to fix the OSRAM Issue. I am quite busy lately and have less time to program than I normally would like to.

**Comment by Vuille** [comment]
OK, I understand. If you tell me a new version compatible with OSRAM  devices will come in a reasonable time, then I will wait and maintain my project in standby.

**Comment by Hyrules** [comment]
If everything goes as planned I should have a release in the next weeks to come. (maybe 1 or 2) I still have one coding issue in the main form that I need more time to fix otherwise the recoded version is better that the previous version.

**Comment by MrHamaga** [comment]
Do you have any issues you want any help with?

**Comment by Hyrules** [comment]
The problem is not with C# itself. It's rather a WPF problem. Trying to bind 2 objects in the main form view so they can hide when not enabled but it seems that I cannot bind a parent to it's children for some reason.

**Comment by Hyrules** [comment]
@Vuille can you do a test for me with the console on the bridge ? I'm curious as how the bridge react when you send a value outside a range to a light. Try sending 500 ct to your lamp 3 which is the Classic B40 TW - Lightify. I want to know if the brige gives an error message or just set the value to the closest max or min with those special cases. The lamp has a specific range for the ct which is different from Philips range. Thank.

**Comment by Vuille** [comment]
Sending 500 ct to light 3 (OSRAM) creates an error. The bridge response is attached.
[test_ct500.txt](https://github.com/Hyrules/WinHue3/files/720542/test_ct500.txt)

**Comment by Vuille** [comment]
Sorry, I did a mistake in my command, please ignore my previous comment. 
Bridge answer to sending 500 ct to light 3 is attached. Ct value does not change.
[test_ct500_2.txt](https://github.com/Hyrules/WinHue3/files/720563/test_ct500_2.txt)

**Comment by Hyrules** [comment]
Ok thanks. This do exactly what I was thinking it does. It goes to the max of this object specs and does not gives an error. I will have a version for you to test this weekend. Thanks again.

**Comment by Vuille** [comment]
It's my pleasure. Please note that I will be absent as of Sunday Jan 22 until Friday Jan 27. So, if you wish me to do other tests, I'll do it tomorrow. Also take your time with the new version, I will not test it before Saturday Jan 28.

**Comment by Hyrules** [comment]
Let me know when you are ready to test I have a version almost ready. I'm currently finishing the rule creator conversion. Just a couple things left to do.

**Comment by Hyrules** [comment]
Here is a version for you to try. Let met know what happens.

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/737849/WinHue.3.BETA.25.Setup.zip)

**Comment by Vuille** [comment]
Thank you.
I will install it and test it this afternoon. Presently, I have one OSRAM lamp but not yet any plug. I plan to buy one shortly.

**Comment by Vuille** [comment]
I have installed that new version. My configuration presently consists of 2 Philips lamps, one Osram light, one occupancy sensor, whereas the following sensors are part of the bridge: daylight, thermostat and hue ambient light.
This is what I have noted:
1) The previous error "bridge not responding" does not occur any longer, except under 8) below
2) I can turn the 3 lamps on and off (double click)
3) Brightness control does not work (can't move the cursor)
4) Daylight sensor responds OK (now it's late afternoon, I have to wait until tomorrow morning to see what it says)
5) Same remark for ambient light sensor
6) Thermostat sensor responds OK
7) Occupancy sensor does not switch from false to true when presence is detected. Besides, it does not give the battery level (field is blank). At the API console, both functions work properly. That was working OK with the previous versions.
8) I had created two simple rules to turn a lamp on and off depending on a presence detected by the occupancy sensor; they workd OK with the previous versions. I had disabled both while waiting for WinHue new version. When I attempted to enable them. I got the error message "bridge not responding (...)" and WinHue closed down. It restarts OK.
9) Apparently, the button Bridge Setting does not work.

I did not test any other functions.
I stay available with pleasure for further tests.

**Comment by Hyrules** [comment]
3. Brightness control depends on the type of light. I hate philips hue api because it's rather an incomplete. From the API a color temperature light should not have a bri settings but from what I see in your last object log you have a bri setting so I will assume that this is the case for all those lights and change the settings so you can adjust the light brightness with the next test I send your.

4. Daylight sensor is a bridge sensor and not a WinHue Sensor same goes for 6.

7. Don't rely on the propertygrid to get the status of the sensor if you don't refresh it yourself. It's not live and does not update unless your click on something else and select it again which force the application to poll the bridge.

8. I will try to fix this in the next test file I send you.

9. Does the window open at all or it just does nothing ? I check the code here and everything seems to be in order here. I will test that at home when I have acces to my bridge.

**Comment by Hyrules** [comment]
@Vuille : Can you give me the json of the said rules ? right click on them in WinHue and select Copy -> JSON formatted and paste it here. It will show me the content of the rule so I can see why it's giving an error. thanks.

**Comment by Vuille** [comment]
3) Brightness control worked with the previous versions of WinHue.
7) It does not update even if I click on something else then come back to it.
8) After I restart WinHue, Rule 1 seems to operate properly, which means the occupancy sensor detects a presence although its status does not indicate it.
9) A error message comes first ("Bridge is not responding") then the window opens but there is no data.

As per your request, this is the JSON of Rule 1:
{
  "name": "presence1",
  "conditions": [
    {
      "address": "/sensors/3/state/presence",
      "operator": "eq",
      "value": "true"
    }
  ],
  "actions": [
    {
      "address": "/lights/1/state",
      "method": "PUT",
      "body": {
        "on": true
      }
    }
  ],
  "owner": "bGSunyXwInVnikD8MrZbGL3ltn9uUWUy7e6ELimp",
  "timestriggered": 0,
  "lasttriggered": "none",
  "created": "2017-01-05T16:42:06",
  "status": "enabled"
}

... and of Rule 2:
{
  "name": "Absence1",
  "conditions": [
    {
      "address": "/sensors/3/state/presence",
      "operator": "eq",
      "value": "false"
    }
  ],
  "actions": [
    {
      "address": "/lights/1/state",
      "method": "PUT",
      "body": {
        "on": false
      }
    }
  ],
  "owner": "bGSunyXwInVnikD8MrZbGL3ltn9uUWUy7e6ELimp",
  "timestriggered": 5,
  "lasttriggered": "2017-01-30T21:27:51",
  "created": "2017-01-05T16:44:54",
  "status": "enabled"
}

**Comment by Hyrules** [comment]
I have a odd bug here with the Motion Sensor... The present is always false... Even if It triggers my rules properly. I'm not speaking of WinHue at all here.

**Comment by Hyrules** [comment]
Let's try thins one. I had forgot the logic that refreshed the object on select. Now implemented. Fix the brightness slider. Try the rules and let me know if there is still a problem. I have not fixed the point 9 as I cannot seem to reproduce it here yet. Can you try reproducing the bridge not responding and send me a log. I will see what's happening.

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/740755/WinHue.3.BETA.25.Setup.zip)

**Comment by Vuille** [comment]
I have installed that new version.
3) Brightness sliders work fine
7) WinHue still does not reflect the state of the occupancy sensor. While I clicked to and fro on other sensors, I got once an error message "Bridge is not responding", without further consequence.
8) Rules work fine, which also means that the occupancy sensor does its job.
9) Bridge settings button works fine.
Now, I have ordered an Osram plug. As soon as I receive it, hopefully within 3-4 days, I will test it and revert to you. Thanks for what you did so far. Let me know if you would like me to do other tests.

**Comment by Hyrules** [comment]
For the occupancy sensor if you want to test it get it beside you and wave your hand over the sensor while clicking back and forth between the sensor and another object. The motion sensor update really fast. If you go in a room and wave at it so it goes to true it will revert back to false a soon a you leave the room. I have tested it here and the object updates properly. I think I see a pattern in your Bridge Not Responding message. I will try to fix it.

**Comment by tursoe** [comment]
Great with sorting of light/groups.
Unfortunately the bulb is no longer the same width as they were before.

Another question, can you under copy rules ask for the new sensor and group/bulb for the command. If I'll copy a rule and want to use another sensor e.g. dimmer switch then the is must be changed. It's the same problem with the actions, it could be fine if it's possible to change affected bulb/group.

I'll keep testing the new version for bugs/improvements.

Best regards
Tursoe

**Comment by tursoe** [comment]
Another strange thing, Resource Link missing the same sort procedure as main window and it's shown horizontal instead of vertical.

**Comment by Hyrules** [comment]
The problem with the bulb is that the text wrap is uneven if I make the bulb all the same size. It's not a bug per se I find it better that way.

For the rule copy it's the reason I have added the copy and edit. The problem with the rules is that if there are 2 different sensors how can I know which sensor to replace ? Adding another window would remove the effect of quick copy no ? The resource link creator does not implement the same way as the main window. I have not added the possibility to sort it. The windows display the objects the same way as the main windows.

**Comment by tursoe** [comment]
Great, I'll also notice that the sorting settings are not saved between startups.

> The problem with the bulb is that the text wrap is uneven if I make the bulb all the same size. It's not a bug per se I find it better that way.
> 
Ok, I'll prefer the old view even it wrap.

> For the rule copy it's the reason I have added the copy and edit. The problem with the rules is that if there are 2 different sensors how can I know which sensor to replace ? Adding another window would remove the effect of quick copy no ?
> 
Your right but an easy way to edit the affected bulb/groups would be nice. As for now, it highlight an action you cannot change type and object.

> The resource link creator does not implement the same way as the main window. I have not added the possibility to sort it. The windows display the objects the same way as the main windows.
> 
Then there is a wrong view on my computer.
There are only horizontal within the types.

**Comment by Hyrules** [comment]
For the resource link i meant that visualy it's the same as the mainform but the code is not the same. I have not implemented sorting. as for the settings and sorting it is not yet saved to the settings and not persistent. I will add this and for the wrapping I will also add the possibility to wrap it as a setting.

**Comment by mcmicha** [comment]
The last WinHue 3 BETA 25 setup works for me so far.
But there is a crash with me, if I want to edit a certain Resoure Links.
The other two Resoure Links can be opened in the editor.
[WinHue3 2017-02-02.txt](https://github.com/Hyrules/WinHue3/files/749214/WinHue3.2017-02-02.txt)

**Comment by Hyrules** [comment]
if i'm not mistaken it's because of /group/0 because group 0 does not really exists as an object on the bridge although it is.

**Comment by Vuille** [comment]
I encountered difficulties while editing a schedule created under earlier versions of WinHue. Frequent error messages "Bridge not responding" and WinHue closed down. So I deleted that schedule. From that moment, the "create a schedule" button turned grey and I could no longer create one. So I did it successfully on the bridge API. I restarted WinHue and the recently created schedule appeared on the schedules pane, but impossible to edit it.

**Comment by Hyrules** [comment]
Schedules are stored on the bridge so they are complient with Philips Hue. In order to schedule something you need to select an object on the main view. You can schedule a group, a light or a scene. the button wil be enabled only if you select one of those.

**Comment by Hyrules** [comment]
For everyone information BETA 25 is not officially released. So some part of the program like scene might not work. I suggest you use BETA 24 if you want everything to mostly work properly.

**Comment by tursoe** [comment]
Another missing element in beta 25.
I'm not able to delete actions when I'm edit rules. I can add new actions but not remove old ones.

**Comment by Hyrules** [comment]
Corrected. The binding was not pointing at the right place.

**Comment by Hyrules** [comment]
Here is a completely functionning WinHue 3 BETA 25. Let me know if you find any bug in any function. If you happen to find an exception. Try use procdump included (see the help tab) and capture the exception in a dump. By clicking on procdump a shell window will open to monitor WinHue. Beware dumps can be big ( 300 - 400 mb) otherwise you can send me the log and i'll try to fix the bug. The dump will be located in the %APPDATA%\WinHue3 folder. I will give you a link to upload a dump in any case you want to send one to me. It's really easier to debug with a dump than with the log because I see the code and the state of the variables and memory a the point the program crash.

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/753421/WinHue.3.BETA.25.Setup.zip)

**Comment by Vuille** [comment]
I have installed the new version, but impossible to launch it. When I double click on the desktop icon, an error message appears and the program closes down. I have restarted the whole process without success. A log file is attached.
[WinHue3 2017-02-06.txt](https://github.com/Hyrules/WinHue3/files/754043/WinHue3.2017-02-06.txt)

**Comment by Hyrules** [comment]
My Bad... I made some cleanup and somehow forgot to add an reference to avalondock in inno. I will fix this later today.

**Comment by Hyrules** [comment]
Here this version should work properly.

[WinHue 3 BETA 25 Setup.zip](https://github.com/Hyrules/WinHue3/files/755889/WinHue.3.BETA.25.Setup.zip)

**Comment by Vuille** [comment]
Thank you, installed it OK. Everything works fine.
Now I wish to integrate an OSRAM plug in the system. I intend to use it to control non-hue lights and devices (just switch on/off, no fading). How shall I do? Is there a pane for plugs or switches in WinHue, as there are for lights and sensors? How to search for that plug?
Note that I did not find in the literature a way to control that plug with the bridge API.

**Comment by Hyrules** [comment]
Do as you would do for a normal bridge object. If the bridge detects it it should appear in WinHue. I think the plug will appear as light. So you can control them with the double click. Searching should be straight forward. Click on search new light. it will take a minute for the bridge to find new object and then it will appear in WinHue.

**Comment by Hyrules** [comment]
Steps should be as following : (taken here https://developers.meethue.com/content/osram-lightify-philips-hue)

1. Power off the hue bridge.

2. Turn off all other smart bulbs (as in turn the plugs/switches off to them) in the house (I have a mix of hue colour, hue lux, livingwhites and the osram bulbs)

3. Set-up lightify gateway (obviously) and apply the wifi and zifbee updates it's prompts for.

4. Reset the bulb I want to update by switching it on for 5 seconds then off for 5 seconds and repeat that 5 times. After turning it on for the fifth time leave it on and it will blink to confirm it has reset.

5. Power off the newly reset osram

6. Add it to the lightify gateway via the app

7. Update it to the latest firmware (takes a while)

8. Unplug the lightify gateway

9. Reset the osram bulb again then switch it off

10. Plug the hue bridge back in and let it power up

11. Search for new bulbs and immediately after pressing search turn on the osram bulb (timing can be tricky and took a couple of goes on two of the bulbs)

12. Done!

**Comment by Hyrules** [comment]
@tursoe : Looking at the cloning feature you are requesting and having some questions. What should the popup window as to replace ? The object selected in the conditions ?

**Comment by mcmicha** [comment]
This [Resource crash](https://github.com/Hyrules/WinHue3/issues/51#issuecomment-277119623) is eliminated.
The new WinHue version found the "Group 0".

**Comment by Vuille** [comment]
WinHue does not discover the Osram plug. I tried on the bridge API as well, without success.
Now, I have a hard time setting up the Osram gateway, in order to reset the plug as per the procedure you indicated. The process is cumbersome; there is no Windows app so I do it on my Android phone. I'll keep you posted.

**Comment by Hyrules** [comment]
Yeah it seems like a pain to get the bridge detect the osram devices. l'll await your input on this.

**Comment by Hyrules** [comment]
Closing this issue. You can open a new one when you are ready.

**Comment by tursoe** [comment]
Hi, I'm just now back from a business trip.
I will look at it within some hours and reply quick.
Christian
